File: cpp_solutions/260.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        long long bitmask = 0;
        for (auto& num : nums) bitmask ^= num;
        
        // lsb
        long long diff = bitmask & (-bitmask);
        long long x = 0;
        for (auto& num : nums) {
            if (num & diff) x ^= num;
        }
        return {static_cast<int>(x), static_cast<int>(bitmask ^ x)};
    }
};

File: rust_solutions/260.rs
impl Solution {
    pub fn single_number(nums: Vec<i32>) -> Vec<i32> {
        let mut value = 0;
        for &num in &nums {
            value ^= num;
        }
        let diff = value & (-value);
        let mut x = 0;
        for &num in &nums {
            if diff & num != 0 {
                x ^= num;
            }
        }
        vec![value ^ x, x]
    }
}
